packages/artifact/__tests__/util.test.ts

```diff
@@ -216,12 +227,43 @@ describe('Utils', () => {
     const directory2 = path.join(directory1, 'folder1')
 
     // Initially should not exist
-    expect(fs.existsSync(directory1)).toEqual(false)
-    expect(fs.existsSync(directory2)).toEqual(false)
+    await expect(fs.promises.access(directory1)).rejects.not.toBeUndefined()
+    await expect(fs.promises.access(directory2)).rejects.not.toBeUndefined()
     const directoryStructure = [directory1, directory2]
     await utils.createDirectoriesForArtifact(directoryStructure)
     // directories should now be created
-    expect(fs.existsSync(directory1)).toEqual(true)
-    expect(fs.existsSync(directory2)).toEqual(true)
+    await expect(fs.promises.access(directory1)).resolves.toEqual(undefined)
+    await expect(fs.promises.access(directory2)).resolves.toEqual(undefined)
+  })
+
+  it('Test Creating Empty Files', async () => {
+    const root = path.join(__dirname, '_temp', 'empty-files')
+    await io.rmRF(root)
+
+    const emptyFile1 = path.join(root, 'emptyFile1')
+    const directoryToCreate = path.join(root, 'folder1')
+    const emptyFile2 = path.join(directoryToCreate, 'emptyFile2')
+
+    // empty files should only be created after the directory structure is fully setup
+    // ensure they are first created by using the createDirectoriesForArtifact method
+    const directoryStructure = [root, directoryToCreate]
+    await utils.createDirectoriesForArtifact(directoryStructure)
+    await expect(fs.promises.access(root)).resolves.toEqual(undefined)
```

Comment:
For asynchronously checking if files exist, I used [this](https://stackoverflow.com/questions/4482686/check-synchronously-if-file-directory-exists-in-node-js) for guidance

Reply:
What's wrong with `fs.existsSync`?

Reply:
`fs.existsSync` is perfectly fine however I'm trying to use async/promises wherever possible. In general any sync operation will block the event loop until the operation is complete so asyncification is usually a good thing.